### PR TITLE
Documentation fix in AboutCsvCmdlets and code fix in AboutXml to fix XML file load issue.

### PR DESCRIPTION
**Pull request by dgoldman-msft**
# PR Summary

In this PR I made one small documentation change in AboutCsvCmdlets for a clearer definition during one test and a fix in AboutXml to avoid creating a blank XML document which breaks the module.

## Context

I spoke to Joel with regards to these changes.

## Changes

1. Change 1 in AboutCsvCmdlets - Added note to make it clear the delimiter needs to be changed by the end user for successful tests
2. Change 2 in AboutXml - Adding BeforeAll statement to create TestDrive path to avoid Show-Karma creating a blank xml document breaking the module.

## Checklist

- [X ] Pull Request has a meaningful title.
- [X ] Summarized changes.
- [X] Pull Request is ready to merge & is not WIP.
- [ ] Added tests / only testable interactively.
  - Make sure you add a new test if old tests do not effectively test the code changed.
- [ ] Added documentation / opened issue to track adding documentation at a later date.

### CODE REVIEW

File: PSKoans/Koans/Constructs and Patterns/AboutXml.Koans.ps1 [hunk 1]
```diff
@@ -11,7 +11,10 @@ param()
     in the examples which follow, distinguishing the approach from reliance on XPath.
 #>
 Describe 'About XML' {
-
+	BeforeAll 	{
+		$path = Join-Path $TestDrive -ChildPath document.xml 
+	}
```
[the remark this answers is not in the document]
Comment (dgoldman-msft) [reply PSKoans/Koans/Constructs and Patterns/AboutXml.Koans.ps1:16]: I was debugging an issue where running Show-Karma and without the BeforeAll statement we would hit line 42 and run the following code and replace the live file breaking the Kona test. Joel had me test that code implementation just for this file which fixes the problem. I could run a test purging the other Join-Paths to see if they are needed. Joel didn't mention at the time of implementation.

 Set-Content -Path $path -Value @'
<documentRoot>
    <element>A short XML file</element>
</documentRoot>
'@

**Comment by dgoldman-msft** [comment]
That makes sense 😊 I am able to reproduce live on my system if I remove the BeforeAll statement. What will happen is the following:


  1.  Open Code and load up the AboutXML.ps1
  2.  In PS or Windows Terminal run: Show-Karma -topic AboutXML or Show-Karma

Expected result is to have a successful test run.
Actual result :  The AboutXml.ps1 file is changed on the fly to just the set-content from like 41 breaking the show-karma.

[00:03:58] (8) ..\0.67.1\public> show-karma
Invoke-Koan : No modules were removed. Verify that the specification of modules to remove is correct and those modules
exist in the runspace. At C:\Users\sifu1\OneDrive\Documents\WindowsPowerShell\Modules\pskoans\0.67.1\Public\Get-Karma.ps1:111 char:32
+                 $PesterTests = Invoke-Koan @{
+                                ~~~~~~~~~~~~~~
    + CategoryInfo          : ResourceUnavailable: (:) [Invoke-Koan], InvalidOperationException
    + FullyQualifiedErrorId : Modules_NoModulesRemoved,Invoke-Koan

Invoke-Koan : No modules were removed. Verify that the specification of modules to remove is correct and those modules
exist in the runspace. At C:\Users\sifu1\OneDrive\Documents\WindowsPowerShell\Modules\pskoans\0.67.1\Public\Get-Karma.ps1:111 char:32
+                 $PesterTests = Invoke-Koan @{
+                                ~~~~~~~~~~~~~~
    + CategoryInfo          : ResourceUnavailable: (:) [Invoke-Koan], InvalidOperationException
    + FullyQualifiedErrorId : Modules_NoModulesRemoved,Invoke-Koan

Invoke-Koan : Cannot find drive. A drive with the name 'TestDrive' does not exist. At C:\Users\sifu1\OneDrive\Documents\WindowsPowerShell\Modules\pskoans\0.67.1\Public\Get-Karma.ps1:111 char:32
+                 $PesterTests = Invoke-Koan @{
+                                ~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (TestDrive:String) [New-Item], DriveNotFoundException
    + FullyQualifiedErrorId : DriveNotFound,Microsoft.PowerShell.Commands.NewItemCommand
    + PSComputerName        : localhost

Invoke-Koan : The term 'console-write' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again. At C:\Users\sifu1\OneDrive\Documents\WindowsPowerShell\Modules\pskoans\0.67.1\Public\Get-Karma.ps1:111 char:32
+                 $PesterTests = Invoke-Koan @{
+                                ~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (console-write:String) [Invoke-Koan], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException
    + PSComputerName        : localhost

Invoke-Koan : Cannot find drive. A drive with the name 'TestDrive' does not exist. At C:\Users\sifu1\OneDrive\Documents\WindowsPowerShell\Modules\pskoans\0.67.1\Public\Get-Karma.ps1:111 char:32
+                 $PesterTests = Invoke-Koan @{
+                                ~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (TestDrive:String) [New-Item], DriveNotFoundException
    + FullyQualifiedErrorId : DriveNotFound,Microsoft.PowerShell.Commands.NewItemCommand
    + PSComputerName        : localhost

Invoke-Koan : The term 'console-write' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again. At C:\Users\sifu1\OneDrive\Documents\WindowsPowerShell\Modules\pskoans\0.67.1\Public\Get-Karma.ps1:111 char:32
+                 $PesterTests = Invoke-Koan @{
+                                ~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (console-write:String) [Invoke-Koan], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException
    + PSComputerName        : localhost

I was able to track this down through debugging and then reached out to Joel. Do you not have the same issue? Subsequently I just ran a test the removal of all the other Join-Paths and this also works for a success test run.

Thanks
Dave

File: PSKoans/Koans/Constructs and Patterns/AboutXml.Koans.ps1 (hunk 1 above)
[the remark this answers is not in the document]
Comment (vexx32) [reply PSKoans/Koans/Constructs and Patterns/AboutXml.Koans.ps1:16]: Looks like we're using tab spacing here at the moment? Spaces only please, finding stray whitespace can be a pain later.

**Review by vexx32 — CHANGES_REQUESTED** [review]
Looks good to me, with one whitespacing thing to sort out before we merge. 🙂

**Review by vexx32 — APPROVED** [review]
Thanks! 😁

**Pull request merged**